File: ui/src/app/models/user-footmatch.ts
import { Column, Entity, ManyToOne } from 'entitype';

import { Footmatch } from './footmatch';
import { User } from './user';

@Entity('user_footmatch')
export class UserFootmatch {

  @Column({ columnName: `user_id`, type: `int(11)`, nullable: false, primaryKey: true, default: null })
  _userId: number;

  @Column({ columnName: `match_id`, type: `int(11)`, nullable: false, primaryKey: true, default: null })
  _matchId: number;

  @Column({ columnName: `user_goal`, type: `int(11)`, default: null })
  _userGoal?: number;

  @ManyToOne(type => UserFootmatch, x => x._matchId)
  _match: Footmatch;

  @ManyToOne(type => UserFootmatch, x => x._userId)
  _user: User;


  public get userId(): number {
    return this._userId;
  }

  public set userId(value: number) {
    this._userId = value;
  }

  public get matchId(): number {
    return this._matchId;
  }

  public set matchId(value: number) {
    this._matchId = value;
  }

  public get userGoal(): number {
    return this._userGoal;
  }

  public set userGoal(value: number) {
    this._userGoal = value;
  }

  public get match(): Footmatch {
    return this._match;
  }

  public set match(value: Footmatch) {
    this._match = value;
  }

  public get user(): User {
    return this._user;
  }

  public set user(value: User) {
    this._user = value;
  }
}

File: app/com/models/UserFootmatch.java
package com.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.io.Serializable;
import javax.persistence.*;


/**
 * The persistent class for the user_footmatch database table.
 * 
 */
@Entity
@Table(name="user_footmatch")
@NamedQuery(name="UserFootmatch.findAll", query="SELECT u FROM UserFootmatch u")
public class UserFootmatch implements Serializable {
	private static final long serialVersionUID = 1L;

	@EmbeddedId
	private UserFootmatchPK id;

	@Column(name="user_goal")
	private int userGoal;

	//bi-directional many-to-one association to Footmatch
	@JsonIgnore
	@ManyToOne
	@JoinColumn(name="match_id")
	private Footmatch footmatch;

	//bi-directional many-to-one association to User
	@ManyToOne
	@JoinColumns({
									@JoinColumn(name="user_id", referencedColumnName="user_id"),
									@JoinColumn(name="user_pseudo", referencedColumnName="user_pseudo")
	})
	private User user;

	public UserFootmatch() {
	}

	public UserFootmatchPK getId() {
		return this.id;
	}

	public void setId(UserFootmatchPK id) {
		this.id = id;
	}

	public int getUserGoal() {
		return this.userGoal;
	}

	public void setUserGoal(int userGoal) {
		this.userGoal = userGoal;
	}

	public Footmatch getFootmatch() {
		return this.footmatch;
	}

	public void setFootmatch(Footmatch footmatch) {
		this.footmatch = footmatch;
	}

	public User getUser() {
		return this.user;
	}

	public void setUser(User user) {
		this.user = user;
	}

}